activerecord/lib/active_record/fixtures.rb

```diff
@@ -520,53 +520,64 @@ def self.create_fixtures(fixtures_directory, fixture_set_names, class_names = {}
       }
 
       unless files_to_read.empty?
-        connection.disable_referential_integrity do
-          fixtures_map = {}
+        fixtures_map = {}
 
-          fixture_sets = files_to_read.map do |fs_name|
-            klass = class_names[fs_name]
-            conn = klass ? klass.connection : connection
-            fixtures_map[fs_name] = new( # ActiveRecord::FixtureSet.new
+        fixture_sets = files_to_read.map do |fs_name|
+          klass = class_names[fs_name]
+          conn = klass ? klass.connection : connection
+          fixtures_map[fs_name] = new( # ActiveRecord::FixtureSet.new
               conn,
               fs_name,
               klass,
               ::File.join(fixtures_directory, fs_name))
-          end
+        end
 
-          update_all_loaded_fixtures fixtures_map
+        update_all_loaded_fixtures fixtures_map
 
-          connection.transaction(:requires_new => true) do
-            deleted_tables = Set.new
-            fixture_sets.each do |fs|
-              conn = fs.model_class.respond_to?(:connection) ? fs.model_class.connection : connection
-              table_rows = fs.table_rows
+        all_connections = ActiveRecord::Base.connection_handler.connection_pool_list.map(&:connection)
+        disable_referential_integrity_in_transaction(all_connections) do
+          deleted_tables = Set.new
+          fixture_sets.each do |fs|
+            conn = fs.model_class.respond_to?(:connection) ? fs.model_class.connection : connection
```

Comment:
What happens if User connects to connection A, and we are in a new transaction on connection B?